src/config/clankerTokenV4.ts

```diff
@@ -0,0 +1,385 @@
+      recipients: z
+        .array(
+          z.object({
+            admin: addressSchema,
+            recipient: addressSchema,
+            bps: z.number(),
+          })
+        )
+        .min(1)
+        .refine((v) => v.reduce((agg, cur) => agg + cur.bps, 10_000), {
+          error: 'Recipient amounts must sum to 100%',
+        }),
+    })
+    .optional(),
+  vanity: z.boolean().default(false),
+});
+export type ClankerV4Token = z.input<typeof clankerV4Token>;
+
+export const clankerV4Converter: ClankerTokenConverter<ClankerV4Token> = async (
+  config: ClankerV4Token
+) => {
+  const cfg = clankerV4Token.parse(config);
+
+  if (!cfg.rewards) {
+    cfg.rewards = {
+      recipients: [
+        {
+          admin: cfg.tokenAdmin,
+          recipient: cfg.tokenAdmin,
+          bps: 10_000,
+        },
+      ],
+    };
+  }
+
+  const metadata = stringify(cfg.metadata);
+  const socialContext = stringify(cfg.context);
+
+  const { hook, poolData } = encodeFeeConfig(cfg.fees);
+
+  const { salt, token: expectedAddress } = cfg.vanity
+    ? await findVanityAddressV4(
+        [
+          cfg.name,
+          cfg.symbol,
+          DEFAULT_SUPPLY,
+          cfg.tokenAdmin,
+          cfg.image,
+          metadata,
+          socialContext,
+          BigInt(cfg.chainId),
+        ],
+        cfg.tokenAdmin,
+        '0x4b07',
+        { chainId: cfg.chainId }
+      )
+    : {
+        salt: '0x0000000000000000000000000000000000000000000000000000000000000000',
+        token: undefined,
+      };
+
+  return {
+    address: CLANKER_FACTORY_V4,
+    abi: Clanker_v4_abi,
+    functionName: 'deployToken',
+    args: [
+      {
+        tokenConfig: {
+          tokenAdmin: cfg.tokenAdmin,
+          name: cfg.name,
+          symbol: cfg.symbol,
+          salt: salt as `0x${string}`,
+          image: cfg.image,
+          metadata,
+          context: socialContext,
+          originatingChainId: BigInt(cfg.chainId),
+        },
+        lockerConfig: {
+          locker: CLANKER_LOCKER_V4,
+          rewardAdmins: cfg.rewards.recipients.map(({ admin }) => admin),
+          rewardRecipients: cfg.rewards.recipients.map(({ recipient }) => recipient),
+          rewardBps: cfg.rewards.recipients.map(({ bps }) => bps),
+          tickLower: cfg.pool.positions.map(({ tickLower }) => tickLower),
+          tickUpper: cfg.pool.positions.map(({ tickUpper }) => tickUpper),
+          positionBps: cfg.pool.positions.map(({ positionBps }) => positionBps),
+          lockerData: '0x',
+        },
+        poolConfig: {
+          pairedToken: cfg.pool.pairedToken,
+          tickIfToken0IsClanker: cfg.pool.tickIfToken0IsClanker,
+          tickSpacing: cfg.pool.tickSpacing,
+          hook: hook,
+          poolData: poolData,
+        },
+        mevModuleConfig: {
+          mevModule: CLANKER_MEV_MODULE_V4,
+          mevModuleData: '0x',
+        },
+        extensionConfigs: [
+          // vaulting extension
+          ...(cfg.vault
```

Comment:
In a customer call yesterday they asked if we'd be able to enable multiple vaults for a single token bc they were interested in different lockup schedules. Having the vault/airdrop literally called the vault/airdrop feels a little odd if we add more vaults/airdrop types in the future